### PR TITLE
Add 'debug' exporter

### PR DESCRIPTION
**Pull request by codeboten**
This PR does the following:

* make a copy of exporter/loggingexporter -> exporter/debugexporter
* adds a deprecation notice to loggingexporter
* updates documentation + examples to use `debug` instead of `logging`

I would suggest to move forward w/ this, we could release v0.81.0 w/ both exporters, and then as of v0.82.0, we could only release the debug exporter, but including the logging exporter at v0.81.0 in the release file. Any thoughts?

FIxes https://github.com/open-telemetry/opentelemetry-collector/issues/7769

### CODE REVIEW

**Review by bogdandrutu — COMMENTED** [review]
Alternative, implement a https://github.com/open-telemetry/opentelemetry-collector/blob/main/confmap/converter.go#L12 that converts the config `exporters::logging[/.*]?` to `exporters::debug[/.*]?`. Then we move the code to debug.

**Review by codeboten — COMMENTED** [review]
> Alternative, implement a https://github.com/open-telemetry/opentelemetry-collector/blob/main/confmap/converter.go#L12 that converts the config exporters::logging[/.*]? to exporters::debug[/.*]?. Then we move the code to debug.

@bogdandrutu Two questions around this:
- Would you end up creating a copy of the entire config map to do the replace? I couldn't find a way to delete a key other than by creating another map and omitting it
- This would also require changing any `logging/*` in the configured pipelines, would you still want to go this route?

File: exporter/debugexporter/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Logging Exporter
```
Comment (djaglowski) [inline exporter/debugexporter/README.md:1]:
```suggestion
# Debug Exporter
```

Comment (codeboten) [reply exporter/debugexporter/README.md]: Addressed. I've also removed `loglevel` from the readme as the `debug` exporter may allow us to remove `loglevel` altogether

File: connector/go.mod [hunk 2]
```diff
@@ -4,6 +4,7 @@ go 1.19
 
 require (
 	github.com/stretchr/testify v1.8.4
+	go.opentelemetry.io/collector v0.80.0
```
Comment (jpkrohling) [inline connector/go.mod:7]: Why this change?

File: docs/troubleshooting.md [hunk 3]
```diff
@@ -161,21 +161,21 @@ $ curl -X POST localhost:9411/api/v2/spans -H'Content-Type: application/json' -d
 You should see a log entry like the following from the Collector:
 
 ```
-2020-11-11T04:12:33.089Z	INFO	loggingexporter/logging_exporter.go:296	TraceExporter	{"#spans": 1}
+2020-11-11T04:12:33.089Z	INFO	debugexporter/debug_exporter.go:296	TraceExporter	{"#spans": 1}
```
Comment (jpkrohling) [inline docs/troubleshooting.md:164]: Call me paranoid, but can you bump the date to a date closer to today? It might be just me, but I would be really confused seeing this timestamp and no versions of the collector around that date having this component.

File: exporter/loggingexporter/factory.go [hunk 4]
```diff
@@ -32,9 +32,9 @@ func NewFactory() exporter.Factory {
 	return exporter.NewFactory(
 		typeStr,
 		createDefaultConfig,
-		exporter.WithTraces(createTracesExporter, component.StabilityLevelDevelopment),
-		exporter.WithMetrics(createMetricsExporter, component.StabilityLevelDevelopment),
-		exporter.WithLogs(createLogsExporter, component.StabilityLevelDevelopment),
+		exporter.WithTraces(createTracesExporter, component.StabilityLevelDeprecated),
```
Comment (jpkrohling) [inline exporter/loggingexporter/factory.go:35]: I think the logging exporter will live on for quite a while, right? Would it make sense to get rid of the code for this exporter and embed the new exporter here? You could probably just return the debug exporter's outcome to NewFactory here, no?

Comment (codeboten) [reply exporter/loggingexporter/factory.go:35]: We'll need to release a new version of the exporter to mark it as "deprecated", so i figured i would remove the code on the next release.

I would remove the code completely and continue to include the module in the built distros, WDYT?

File: connector/go.mod (hunk 2 above)
Comment (codeboten) [reply connector/go.mod:7]: i ran `make gotidy` and this happened

File: exporter/loggingexporter/factory.go (hunk 4 above)
Comment (jpkrohling) [reply exporter/loggingexporter/factory.go:35]: > I would remove the code completely and continue to include the module in the built distros, WDYT?

Can you expand on that? Are you going to keep including the same version of the logging exporter? I don't think that's what you meant, but it's not clear to me what the plan is.

Comment (codeboten) [reply exporter/loggingexporter/factory.go:35]: > keep including the same version of the logging exporter?
This would be one option.

Another would be to go w/ the wrapper approach, i originally proposed this. A third option would be to do what @bogdandrutu suggested [here](https://github.com/open-telemetry/opentelemetry-collector/pull/7773#pullrequestreview-1447018212) and implement a converter.

I could go with any of those, but either ways we'll need both modules published for at least one version to deprecate the logging exporter

Comment (jpkrohling) [reply exporter/loggingexporter/factory.go:35]: > Another would be to go w/ the wrapper approach

That's what I also had in mind as well, I think it's the cleanest. The new exporter gets the code, the old exporter wraps the new exporter, so that it's code base is minimal and can continue to exist with almost no maintenance for a year before we delete it from the code base.

> we'll need both modules published for at least one version to deprecate the logging exporter

If the cut off date is July 2024, we should continue publishing new versions of the logging exporter until then. Wrapping the debug exporter within the logging exporter allows that with minimal overhead.

**Comment by codeboten** [comment]
Reopening this PR. Followed the suggestion from @jpkrohling on aliasing the logging exporter as a debug exporter

File: exporter/loggingexporter/factory.go (hunk 4 above)
Comment (codeboten) [reply exporter/loggingexporter/factory.go:35]: Finally getting back to this PR. Updated it to follow your suggestion, PTAL

**Comment by codeboten** [comment]
Well apparently github didn't like me trying to re-open this PR so i opened a new one here: https://github.com/open-telemetry/opentelemetry-collector/pull/8375
